app/assets/javascripts/quby/answers/validation.js

```diff
@@ -189,6 +208,57 @@
     return !failed;
   };
 
+  function parsePartialDate(inputs) {
+    var values = dateValuesWithDefaults(inputs)
+    if(!dateValuesValid(values)) {
+      throw "invalidDate";
+    }
+    
+    return new Date(values.year, values.month, values.day, values.hour, values.minute);
+  }
+
+  function dateValuesValid(values) {
+    return values['year']   >= 1900 && values['year']   <= 2100 &&
+           values['month']  >= 0    && values['month']  <= 11   &&
+           values['day']    >= 1    && values['day']    <= 31   &&
+           values['hour']   >= 0    && values['hour']   <= 23   &&
+           values['minute'] >= 0    && values['minute'] <= 59;
+  }
+
+  function dateValuesWithDefaults(inputs) {
+    var value = function(placeholder) {
+      var val = inputs.filter("[placeholder=" + placeholder + "]").first().val();
+      if(val === undefined || val == "")
+        return null;
+      if(intVal = parseInt(val)) {
+        return intVal;
+      } else {
+        throw "invalidDate";
+      }
+    };
+
+    return {
+      year:   value('YYYY') || 2000,
+      month:  value('MM')   ? value('MM')-1 : 0,    // JS months range from 0-11 instead of 1-12
+      day:    value('DD')   || 1,
+      hour:   value('hh')   || 0,
+      minute: value('mm')   || 0,
+    }
+  }
+
+  function numberOfEmptyDateFields(inputs) {
+    return inputs.toArray().reduce(function(fieldsEmpty, field) {
```

Comment:
Array.reduce is IE9+, helaas kunnen we dat dus nog niet gebruiken. Je kan het beste de underscore `_.map` methode hiervoor gebruiken of anders de jquery map methode.